pwiz_tools/Skyline/Model/SkypFile.cs

```diff
@@ -152,20 +155,50 @@ public bool Open(string skypPath, IEnumerable<Server> servers, FormEx parentWind
             }
             catch (Exception e)
             {
-                var message = TextUtil.LineSeparate(Resources.SkypSupport_Open_Failure_opening_skyp_file_, e.Message);
-                MessageDlg.ShowWithException(parentWindow ?? _skyline, message, e);
-                return false;
+                if (skyp != null && e.Message.Contains(AddPanoramaServerMessage(skyp)))
+                {
+                    return AddServerAndOpen(skypPath, e.Message, parentWindow);
+                }
+                else
+                {
+                    var message = TextUtil.LineSeparate(Resources.SkypSupport_Open_Failure_opening_skyp_file_, e.Message);
+                    MessageDlg.ShowWithException(parentWindow ?? _skyline, message, e);
+                    return false;
+                }
             }
         }
 
+        private bool AddServerAndOpen(string skypPath, String message, FormEx parentWindow)
+        {
+            using (var alertDlg = new AlertDlg(message, MessageBoxButtons.OKCancel))
+            {
+                alertDlg.ShowDialog(parentWindow ?? _skyline);
+                if (alertDlg.DialogResult == DialogResult.OK)
+                {
+                    using (var dlg = new ToolOptionsUI(_skyline.Document.Settings))
+                    {
+                        // Let the user add a Panorama server.
+                        dlg.NavigateToTab(ToolOptionsUI.TABS.Panorama);
+                        dlg.ShowDialog(alertDlg);
+                        if (dlg.DialogResult != DialogResult.OK)
+                        {
+                            return false;
+                        }
+                    }
+                    return Open(skypPath, Settings.Default.ServerList /*get the updated server list*/, parentWindow);
```

Comment:
Does this recurse every time the request fails? Probably not a huge concern, but maybe impose some limit where we stop showing the ToolsOptionsUI and leave the user to actually use Tools > Options as described in the message or try again from scratch.

Reply:
Thanks.  I added a commit limiting retries to 3.

Reply:
I committed a change limiting retries to 3